### ISSUE TITLE
Faces tab is really slow and resource demanding

### ISSUE THREAD

**Issue by faizpuru** [nextcloud/photos#1467]
First I would like to congratulate you for the new version. It's great.

Nextcloud 25.0.1
Photos 2.0.0

Displaying the faces tab is really slow and resource demanding. I can see the grouping is being made incrementally. 
It looks like the grouping by face is made on client. It is currently not usable on a large photo library.

**Comment by gmeks** [comment]
As a heads up, some of that is fixed in my poorly worded case  ( https://github.com/nextcloud/photos/issues/1411 )

**Comment by marcelklehr** [comment]
Yes, this is a duplicate of #1411

**Comment by leegarrett** [comment]
I have tested the tip of master (`332451d`) and can confirm that `/apps/photos/faces` now loads at a reasonable speed.

For any other users reading this: I didn't get the HTTP 504 errors described in #1411. However, the lazy loading patch described there also fixes this issue for me.

The way I tested is fairly simple:

- Download the tarball of this repo
- On the nextcloud instance, move `apps/photos` to `apps/photos~`
- Extract the tarball and move it to `apps/photos`
- Run `make dev-setup && make build-js-production` in `apps/photos`
- I had to edit `apps/photos/appinfo/info.xml` and changed it to `min-version="25"`

Of course if you test the development version and it eats your lunch or kills your cat, you're on your own. :)
